conda/plugins/hookspec.py

```diff
@@ -96,3 +98,30 @@ def conda_virtual_packages():
 
         :return: An iterable of virtual package entries.
         """
+
+    @_hookspec
+    def conda_pre_commands(self) -> Iterable[CondaPreCommand]:
+        """
+        Register pre-commands functions in conda.
+
+        **Example:**
+
+        .. code-block:: python
+
+           from conda import plugins
+
+           PLUGIN_NAME = "custom_plugin"
```

Comment:
```suggestion
```

This is otherwise confusing users between "plugin" and individual hook implementations, one plugin is a collection of hook implementations.